firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -25,67 +26,40 @@
 /** Use file locking to acquire a lock that will also block other processes. */
 class CrossProcessLock {
   private static final String TAG = "CrossProcessLock";
-  private final FileChannel channel;
-  private final FileLock lock;
 
-  private CrossProcessLock(FileChannel channel, FileLock lock) {
-    this.channel = channel;
-    this.lock = lock;
+  interface CrossProcessPersistedInstallation {
+    PersistedInstallationEntry getPersistedInstallations();
   }
 
   /**
-   * Create and return a lock that is exclusive across processes. If another process has the lock
-   * then this call will block until it is released.
+   * Create a lock that is exclusive across processes. If another process has the lock then this
+   * call will block until it is released.
    *
    * @param lockName the lockname is global across all processes of an app
    * @return a CrossProcessLock if success. If the lock failed to acquire (maybe due to disk full or
    *     other unexpected and unsupported permissions) then null will be returned.
    */
-  static CrossProcessLock acquire(Context appContext, String lockName) {
-    FileChannel channel = null;
-    FileLock lock = null;
-    try {
-      File file = new File(appContext.getFilesDir(), lockName);
-      channel = new RandomAccessFile(file, "rw").getChannel();
-      // Use the file channel to create a lock on the file.
-      // This method blocks until it can retrieve the lock.
-      lock = channel.lock();
-      return new CrossProcessLock(channel, lock);
+  static PersistedInstallationEntry whileLocked(
```

Comment:
1. why is this method package visible instead of private?
1. `whileLocked` is not a good method name. I recommend to use
    a. `#getLock` - it's pretty common that if a lambda is given it is executed with whatever the method name described. Example `file.open {f -> f.write(...)}`
    b.`supplySynchronized` - the purpose of this method is to run the given Supplier in a synchronized form.
    c. `supplyWithProcessSync` - ...

It would also be great to get rid of all the other parameters, so that it is clear that this method executes the given `Supplier` in a synchronized context and does nothing else.

Reply:
After thinking more about this, I think a good method name is
`#synchonize`

Then when used it should be self-explanatory:
`crossProcessLock.synchronize( () -> someActionThatNeedsToBeSynchronized() );`

Reply:
Like it. Updated.